libraries/AP_Airspeed/AP_Airspeed_MS4525.cpp

```diff
@@ -22,55 +22,78 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_HAL/I2CDevice.h>
 #include <AP_Math/AP_Math.h>
+#include <GCS_MAVLink/GCS.h>
 #include <stdio.h>
 #include <utility>
 
 extern const AP_HAL::HAL &hal;
 
-#define MS4525D0_I2C_ADDR 0x28
+#define MS4525D0_I2C_ADDR1 0x28
+#define MS4525D0_I2C_ADDR2 0x36
+#define MS4525D0_I2C_ADDR3 0x46
 
 AP_Airspeed_MS4525::AP_Airspeed_MS4525(AP_Airspeed &_frontend, uint8_t _instance) :
     AP_Airspeed_Backend(_frontend, _instance)
 {
 }
 
-// probe and initialise the sensor
-bool AP_Airspeed_MS4525::init()
+// probe for a sensor
+bool AP_Airspeed_MS4525::probe(uint8_t bus, uint8_t address)
 {
-    const struct {
-        uint8_t bus;
-        uint8_t addr;
-    } addresses[] = {
-        { 1, MS4525D0_I2C_ADDR },
-        { 0, MS4525D0_I2C_ADDR },
-        { 2, MS4525D0_I2C_ADDR },
-        { 3, MS4525D0_I2C_ADDR },
-    };
-    bool found = false;
-    for (uint8_t i=0; i<ARRAY_SIZE(addresses); i++) {
-        _dev = hal.i2c_mgr->get_device(addresses[i].bus, addresses[i].addr);
-        if (!_dev) {
-            continue;
-        }
-        WITH_SEMAPHORE(_dev->get_semaphore());
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
```

Comment:
I *think* we're supposed to be canonicalising on checking against `nullptr`.

Personally I'm not fussed either way, but I think we might have pinged people on it before.

Reply:
its an ownptr, which really is bool